patches/chrome-browser-chrome_browser_main.cc.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/browser/chrome_browser_main.cc b/chrome/browser/chrome_browser_main.cc
+index c832319917cb03bc85b416489586480a442dc717..2301f935fffc6f4b4187d655479c2e78261913c0 100644
+--- a/chrome/browser/chrome_browser_main.cc
++++ b/chrome/browser/chrome_browser_main.cc
+@@ -1626,6 +1626,8 @@ int ChromeBrowserMainParts::PreMainMessageLoopRunImpl() {
+   offline_pages::OfflinePageInfoHandler::Register();
+ #endif
+
++  RegisterBraveExtendedInfoHandlers();
```

Comment:
Can we put this into ```BraveBrowserMainParts::PreMainMessageLoopRun``` and don't introduce this patch? ```ChromeBrowserMainParts::PreMainMessageLoopRun``` is virtual.

Reply:
I tried this first.

If I register before, then I miss the call to `sessions::ContentSerializedNavigationDriver::SetInstance(ChromeSerializedNavigationDriver::GetInstance())`. Happens [here](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_main.cc;l=1607).

If I register after calling `ChromeBrowserMainParts::PreMainMessageLoopRun`, it introduces a very subtle bug where the tabs are restored before we register our handler.

Reply:
What about putting it into ```PreBrowserStart```? It's very close to the current position.

Reply:
that worked. patch + chromium_src deleted. thanks :)

Reply:
Would be good if we can `DCHECK` ensuring Chromium doesn't move this logic around. If the registration happens after any serialized navigation entries are loaded, then it's a no-op. A silent bug, the worst.